### PR TITLE
Added a post about our RBAC setup

### PR DESCRIPTION
**Pull request by jer**

### CODE REVIEW

File: _posts/2016-09-13-rbac.markdown [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+## Enabling RBAC in Kubernetes
+
+*Note: In Screwdriver, we have configured our Kubernetes cluster using [Kops](https://github.com/kubernetes/kops). Your configuration may differ from our setup.*
+
+For Kubernetes version 1.3, enabling RBAC mode in Kubernetes is as easy as passing a flag to kube-apiserver:
+
+    --authorization-mode=RBAC
+
+If nothing at works at all after you add the flag, you are most likely on the right track. This so thoroughly disables access that even your admin will be unable to access anything (including roles). For bootstrapping RBAC, you'll need to temporarily give `admin` its priveleges back:
```
Comment (tkyi) [inline _posts/2016-09-13-rbac.markdown]: `If nothing works at all`*

`privileges`

File: _posts/2016-09-13-rbac.markdown [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+# Setting up Roles
+
+Once RBAC is enabled, every aspect of the API is disallowed access. We need to start giving access back. In our first stab, we ended up opening access up a little more than we need. The important part for us now is simply that Screwdriver's user should be able to kick off jobs while the service accounts for those jobs have no access whatsoever.
+
+### Actors
+
+- **admin**: Should have wide-open access to the entire cluster. This allows us to remove the bootstrapping parameter `--authorization-rbac-super-user=admin` from earlier.
+- **kubelet**: The kubelet is every node's interface with Kubernetes, and thus needs to be granted a good amount of access to the cluster. At some point we want to emulate [Openshift's](https://github.com/openshift/origin/blob/master/pkg/cmd/server/bootstrappolicy/policy.go#L565) node policies here. For now, we are giving the kubelet admin access, which lines up with the access that it had pre-RBAC.
+- **sd-build**: This is the ServiceAccount that we use in Screwdriver for kicking off builds. It needs the ability to create and read things inside of the build namespace (currently the `default` namespace)
```
Comment (tkyi) [inline _posts/2016-09-13-rbac.markdown]: missing a period at the end

File: _posts/2016-09-13-rbac.markdown [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+    "/usr/local/bin/kube-apiserver --address=127.0.0.1 --authorization-mode=RBAC --authorization-rbac-super-user=admin --admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,ResourceQuota,PersistentVolumeLabel --allow-privileged=true --basic-auth-file=/srv/kubernetes/basic_auth.csv --client-ca-file=/srv/kubernetes/ca.crt --cloud-provider=aws --etcd-servers-overrides=/events#http://127.0.0.1:4002 --etcd-servers=http://127.0.0.1:4001 --secure-port=443 --service-cluster-ip-range=100.64.0.0/13 --tls-cert-file=/srv/kubernetes/server.cert --tls-private-key-file=/srv/kubernetes/server.key --token-auth-file=/srv/kubernetes/known_tokens.csv --v=2 1>>/var/log/kube-apiserver.log 2>&1"
+
+At this point, you are on the right track if everything is broken except `admin`'s access.
+
+# Setting up Roles
+
+Once RBAC is enabled, every aspect of the API is disallowed access. We need to start giving access back. In our first stab, we ended up opening access up a little more than we need. The important part for us now is simply that Screwdriver's user should be able to kick off jobs while the service accounts for those jobs have no access whatsoever.
```
Comment (tkyi) [inline _posts/2016-09-13-rbac.markdown]: Might be useful to link to some documentation for `service accounts` (http://kubernetes.io/docs/user-guide/service-accounts/)

File: _posts/2016-09-13-rbac.markdown [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+- **sd-build**: This is the ServiceAccount that we use in Screwdriver for kicking off builds. It needs the ability to create and read things inside of the build namespace (currently the `default` namespace)
+- **kube-proxy**: The kube-proxy runs on every node, and is the piece responsible for routing network traffic to the right places for services. From what we can tell, it needs to be able to read (and probably watch?) the kubernetes API, as well as service and pod information. We have configured it for full read access for now, but we're interested in restricting that access further.
+- **kube-system default ServiceAccount**: Things like the Kubernetes UI, InfluxDB, etc. run in the kube-system namespace. We have decided to open cluster-level access up for the default ServiceAccount in this namespace. We will save fine-grained access for a future date.
```
Comment (tkyi) [inline _posts/2016-09-13-rbac.markdown]: - capitalize Kubernetes
- put backticks around kube-system (`kube-system` namespace) for consistency

**Comment by tkyi** [comment]
LGTM 👍 
giphy-3

**Comment by stjohnjohnson** [comment]
Seems reasonable 👍  I mean, you're no @Filbird when writing blog posts.

giphy 1

**Comment by FenrirUnbound** [comment]
👍 

Looks good to me

mercedes benz concept iaa - imgur

**Comment by Filbird** [comment]
I am humbled. Looking good :)

**Pull request merged**